### PR TITLE
avoid crashing in removeReturns for BlockStatements

### PR DESCRIPTION
**Pull request by ChrisDodd**
- BlockStatements in the initializers of Delcaration_Instances, whicha re not under controls or actions.

### CODE REVIEW

File: midend/removeReturns.cpp [hunk 1]
```diff
@@ -132,7 +132,8 @@ const IR::Node* DoRemoveReturns::preorder(IR::BlockStatement* statement) {
             ret = r;
         }
     }
-    set(ret);
+    if (!stack.empty())
```
Comment (mbudiu-vmw) [inline midend/removeReturns.cpp]: Can you add a comment indicating why an empty stack is OK here?

Comment (ChrisDodd) [inline midend/removeReturns.cpp]: Its not entirely clear that it is OK -- perhaps it should be pruning the preorder function for Delcaration_Instance instead.

**Comment by ChrisDodd** [comment]
added testcase

File: midend/removeReturns.cpp (hunk 1 above)
Comment (mbudiu-vmw) [inline midend/removeReturns.cpp]: Or maybe just 
```
    bool preorder(const IR::Function*) override
    { return false; }
```

**Pull request merged**